plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3/TrinoS3FileSystem.java

```diff
@@ -1670,6 +1671,10 @@ private void finishUpload(String uploadId)
 
         private void abortUpload()
         {
+            if (aborted) {
+                return;
+            }
+            aborted = true;
```

Comment:
should `failed` be true now too?

Reply:
will drop `abort` change for now to keep this PR focused.